src/coreclr/pal/inc/unixasmmacrosamd64.inc

```diff
@@ -366,3 +372,45 @@ C_FUNC(\Name\()_End):
         .cfi_same_value rbp
 
 .endm
+
+// Inlined version of GetThreadEEAllocContext. Trashes volatile registers.
+.macro INLINE_GET_ALLOC_CONTEXT
+#if defined(FEATURE_EMULATED_TLS) || defined(__APPLE__)
+        call    C_FUNC(GetThreadEEAllocContext)
+#else
+        .att_syntax
+        .byte 0x66  // data16 prefix - padding to have space for linker relaxations
+        leaq    t_runtime_thread_locals@TLSGD(%rip), %rdi
+        .byte 0x66  //
+        .byte 0x66  //
+        .byte 0x48  // rex.W prefix, also for padding
+        callq   __tls_get_addr@PLT
+        .intel_syntax noprefix
+
+        .ifnc OFFSETOF__RuntimeThreadLocals__ee_alloc_context, 0
+                lea     rax, [rax + OFFSETOF__RuntimeThreadLocals__ee_alloc_context]
+        .endif
+#endif
+.endm
+
+// Pushes a TransitionBlock on the stack without saving the FP argument registers.
+// See the PROLOG_WITH_TRANSITION_BLOCK macro for the stack layout.
+.macro PUSH_COOP_PINVOKE_FRAME target
+        set_cfa_register rsp, 8
+
+        __PWTB_StackAlloc = 8 // alignment of the stack to 16b
+        __PWTB_TransitionBlock = __PWTB_StackAlloc
+
+        PUSH_CALLEE_SAVED_REGISTERS
+        PUSH_ARGUMENT_REGISTERS
```

Comment:
Are the argument registers actually used for anything? Should this just be a large StackAlloc instead?

Reply:
Pushed the change for all 64-bit platforms. It's more involved on x86 where we rely on the saved arguments as a side-effect (likely fixable but needs more elaborate approach and testing). On ARM32 it's a single instruction to save/restore the arguments so it's not likely to have any reasonable effect compared to the rest of the code around.